common/types.go

```diff
@@ -168,3 +169,40 @@ type LogConfig struct {
 	FileCount   uint64
 	TimeTracker bool
 }
+
+type Block struct {
+	Id         string
+	StartIndex int64
+	EndIndex   int64
+	Size       int64
+}
+
+// list that holds blocks containing ids and corresponding offsets
+type BlockOffsetList []*Block
+
+func (bol BlockOffsetList) FindBlocksToModify(offset, length int64) (BlockOffsetList, int64, bool) {
+	// size of mod block list
+	size := int64(0)
+	currentBlockOffset := offset
+	var modBlockList BlockOffsetList
+	// TODO: change this to binary search (logn) for better perf
+	for _, blk := range bol {
+		if currentBlockOffset >= blk.StartIndex && currentBlockOffset <= blk.EndIndex && currentBlockOffset <= offset+length {
+			modBlockList = append(modBlockList, blk)
+			size += blk.Size
+			currentBlockOffset = blk.EndIndex
+		}
+	}
+	return modBlockList, size, offset+length >= bol[len(bol)-1].EndIndex
```

Comment:
Can you add a comment for what the bool return is for? Just for easy reading

Reply:
This is if we need to create more blocks right?

Reply:
done